src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/PagedBufferedTextWriter.cs

```diff
@@ -83,129 +83,146 @@ public override void Write(char value)
 
         public override void Write(char[] buffer)
         {
+            if (buffer == null)
+            {
+                return;
+            }
+
             Write(buffer, 0, buffer.Length);
         }
+    }
 
-        public override void Write(char[] buffer, int index, int count)
+    public override void Write(char[] buffer, int index, int count)
+    {
+        if (buffer == null)
         {
-            while (count > 0)
-            {
-                var page = GetCurrentPage();
-                var copyLength = Math.Min(count, page.Length - _currentIndex);
-                Debug.Assert(copyLength > 0);
-
-                Array.Copy(
-                    buffer,
-                    index,
-                    page,
-                    _currentIndex,
-                    copyLength);
-
-                _currentIndex += copyLength;
-                index += copyLength;
-
-                count -= copyLength;
-            }
+            throw new ArgumentNullException(nameof(buffer));
```

Comment:
@rynowak could you to confirm that having this throw is the correct behavior? It seems pretty dangerous because it causes the response to fail after we've already sent the headers but @dougbu is correct that that's what we do in `System.IO.TextWriter`.